src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/JsonReaderExtensions.cs

```diff
@@ -1,78 +1,35 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-#nullable disable
-
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
 using Newtonsoft.Json;
 
 namespace Microsoft.AspNetCore.Razor.ProjectEngineHost.Serialization;
 
 internal static class JsonReaderExtensions
```

Comment:
❓Is there a real reason not to just add these to JsonReader directly?

Reply:
`JsonReader` is from `JSON.NET`. `JsonDataReader` is our wrapper, but it isn't always used. `ObjectJsonConverter<T>` calls these methods directly on `JsonReader` before creating a `JsonDataReader`.